### PR TITLE
Failing test for <Miss /> with 'Blocker' component

### PR DESCRIPTION
**Pull request by alisd23**
Refers to issue #4035

### CODE REVIEW

File: modules/__tests__/Miss-test.js [hunk 1]
```diff
@@ -94,8 +108,42 @@ describe('Miss', () => {
       render(<App location={nestedLoc} />, div, () => {
         expect(div.innerHTML).toContain(TEXT)
         expect(div.innerHTML).toNotContain(MATCH)
+        unmountComponentAtNode(div)
         done()
       })
     })
   })
+
+  describe('FAILING MISS TESTS', (done) => {
```
Comment (pshrmn) [inline modules/__tests__/Miss-test.js]: `done` should be an argument to the `it` callback, not `describe`

**Comment by ryanflorence** [comment]
alright, I'm digging in on this today

**Comment by ryanflorence** [comment]
afaict Miss works fine on master since we added the `<LocationSubscriber>` to it, I think the test is trolling you, if you click around the examples it works fine. Feel free to reopen if you find something, I'm just cleaning house here.

**Comment by alisd23** [comment]
Updated to current v4 master, and test still fails. Same issue with `<Blocker>` causing `removeMatch` to be called too late, which means `notifySubscribers` has already fired with the incorrect match information and <Miss>` has already re-rendered.

Mentioned in more detail in #4035.
This PR fixes the issue I believe - #4067

**Comment by ryanflorence** [comment]
I know the test fails, but if you click around the examples w/ a `<Blocker/>` it still works. Can you confirm?

**Comment by pshrmn** [comment]
Try this app adapted from the test. When you navigate directly to `/` or `/other`, they render as expected, but when clicking the links they don't.

```js
import React from 'react'
import { Match, Miss, Link, HashRouter } from 'react-router'

class Blocker extends React.Component {
  shouldComponentUpdate() {
    return false
  }

  render() {
    return <App />
  }
}

const Home = () => (
  <Link to='/other'>Other</Link>
)

const NotFound = () => (
  <div>
    <Link to='/'>Home</Link>
    <p>Not Found</p>
  </div>
)

const App = () => (
  <div>
    <Match pattern='/' exactly component={Home} />
    <Miss component={NotFound} />
  </div>
)

export default () => (
  <HashRouter>
    <Blocker />
  </HashRouter>
);
```

**Comment by alisd23** [comment]
Test still failing when up to date in `v4` branch (alpha-5)

**Comment by pshrmn** [comment]
FWIW this isn't an issue with `<MatchGroup>` :wink:

**Comment by mjackson** [comment]
I'm afraid this PR is a bit outdated. `<Miss>` is gone in current v4, huzzah!

**Comment by alisd23** [comment]
:+1:

**Pull request closed without merging**